src/mbgl/style/sources/geojson_source_impl.cpp

```diff
@@ -67,23 +67,55 @@ class SuperclusterData : public GeoJSONData {
     mapbox::supercluster::Supercluster impl;
 };
 
-GeoJSONSource::Impl::Impl(std::string id_, GeoJSONOptions options_)
-    : Source::Impl(SourceType::GeoJSON, std::move(id_)),
-      options(std::move(options_)) {
+template <class T>
+T EvaluateFeature(optional<T> accumulated,
+                  mapbox::feature::feature<double>& f,
+                  const std::shared_ptr<expression::Expression> expression,
+                  T finalDefaultValue = T()) {
+    const expression::EvaluationResult result = expression->evaluate(accumulated, f);
+    if (result) {
+        const optional<T> typed = expression::fromExpressionValue<T>(*result);
+        return typed ? *typed : finalDefaultValue;
```

Comment:
std::move(*typed)

Reply:
Maybe would be better if:
```c++
if (result) {
   optional<T> typed = expression::fromExpressionValue<T>(*result);
   if (typed) {
     return *typed;
   }
}
return T();
```

Reply:
If we want to use move semantic, then `typed` can not be const. As based on Effective Morden C++

> First, don’t declare objects const if you want to be able to move from them. Move requests on const objects are
silently transformed into copy operations